tagsets/cronjobs.php

```diff
@@ -448,7 +448,7 @@ function cron__check_for_finished_experiments(){
     while ($line=pdo_fetch_assoc($result)) {
         // if the last session start time is more than 4 weeks ago the experiment is considered to be finished
         $session_start = ortime__sesstime_to_unixtime($line['session_start']);
-        if ($session_end + $four_weeks < $now) {
+        if ($session_start + $four_weeks < $now) {
```

Comment:
how did it work before this change? I don't get it. Is the only change here that the time to auto-finish the experiment is 4 weeks after the _beginning_ of the last session instead of the _end_ of the last session?

Reply:
Actually where were two bugs:

1. In line 451 the uninitialized variable `$session_end` was used. Having a uninitialized variable as an operand doesn't result in an error. So `$session_end + $four_weeks` is equal to `$four_weeks` which is (almost ever) smaller than `$now`.

2. The line 250 in combination with line 437. Before this fix the array `$line` didn't have a key `session_start` because the sql server returned a key name `max(session_start)` for the second column of the result table. Accessing a undefined index in a PHP array is not an error. It will return NULL. `ortime__sesstime_to_unixtime` uses `substr()` and `mktime()` internally. `substr()` on a NULL will return false and I don't know exactly what `mktime()` will do with false as input value but it will output some number...